libmscore/instrtemplate.cpp

```diff
@@ -291,11 +293,15 @@ void InstrumentTemplate::write(XmlWriter& xml) const
             xml.tag("drumset", int(useDrumset));
       if (drumset)
             drumset->save(xml);
-      foreach(const NamedEventList& a, midiActions)
+      
+      if (!singleNoteDynamics)      // default is true
```

Comment:
Are there so many instruments that support single-note dynamics change compared to those that don't support it that we should make it a default? Maybe it would be better to always write this value explicitly.

Reply:
Yes, there are a fair few more. It's about 190 (non-expression) vs 360 (expression). Although, I guess we could always write this explicitly. In instruments.xml, it's only specified if it's _non-expressive_. Also, I thought one of our design principles was to only write values when they're not the default?

Reply:
So am I OK to leave it as it is, or do you still think that writing it always explicitly always is the way to go?

Reply:
Yes, I believe it is OK to keep the current choice, at least unless there are some other opinions on that.

Reply:
I think keep it in. The abstract case should be the most general (i.e. include every possible way of making a sound), and then specific instruments opt-out of features they don't need or can't use. Also, there may case where the user may want to override the default setting for artistic reasons even if the real instrument is not capable of producing that sound.